### DISCUSSION TITLE
Osu std Fun category mod "4 key" proposal

### DISCUSSION THREAD

**Discussion by LosPedros** [ppy/osu#18024] [Ideas]
Recently I've been trying for some reason to alt with 4 keys (2 on the keyboard and 2 on the mouse) and I'm really enjoying it even if I can only play 4/5*, but its been bugging me that mouse1 key1 and mouse2 key2 are being treated as one input, my proposal is to add a mod in the Fun category that would treat mouse keys as separate inputs from the keyboard ones

https://user-images.githubusercontent.com/46714621/166064852-34de20bf-49a6-4434-8ee6-3dfd50ff8f40.mp4

**Comment by peppy** [comment]
In case extra context is required: the fact that both keys assigned to the same game "button" are the same overlapping input is intended. This is to avoid the huge advantage offered by adding extra keys to the game (which would break existing leaderboards in some cases).

Adding a mod sounds interesting, but also potentially a bit difficulty due to how key bindings are handled.

**Comment by Blobanium** [comment]
Adding a 4 key mod might be a good idea, especially with the somewhat new controversial raketapping method that has been plaguing the osu!std meta. Going to quote peppy's comment just for reference.

>  This is to avoid the huge advantage offered by adding extra keys to the game (which would break existing leaderboards in some cases).

We could add a score nerf (as well as nerf star rating and performance points) to mods that allow the usage of more than 2 buttons in order to prevent this from happening. My proposal is 0.5x Score Multiplier for 4k (and 0.67x if you want to implement a 3k mod for osu!std as well.)


> Adding a mod sounds interesting, but also potentially a bit difficult due to how key bindings are handled.

instead of M1 and K1 being a single input during, I'd rather utilize a K1, K2, K3, and K4 as button layout (kind of like mania but in osu!std) just to make things more simple and easier to implement.

**Comment by shiumano** [comment]
What happens in the case of touch panels?
(Already, it is possible to spam with three fingers.)

**Reply by LosPedros** [reply]
Im guessing it could probably not be allowed with TD just like you cant have DT and NC at the same time

**Reply by Blobanium** [reply]
Like lospedros said touch device mods would probably would be incompatable with eachother
